rs/ic_os/network/src/systemd.rs

```diff
@@ -120,33 +120,43 @@ pub fn generate_systemd_config_files(
     ipv6_address: &Ipv6Addr,
 ) -> Result<()> {
     let mut interfaces = get_interfaces()?;
-    interfaces.sort_by(|a, b| a.speed_mbps.cmp(&b.speed_mbps));
+    interfaces.sort_by(|a, b| b.speed_mbps.cmp(&a.speed_mbps));
     eprintln!("Interfaces sorted by speed: {:?}", interfaces);
 
     let ping_target = network_info.ipv6_gateway.to_string();
-    // Old nodes are still configured with a local IPv4 interface connection
-    // Local IPv4 interfaces must be filtered out
-    let ipv6_interfaces: Vec<&Interface> = interfaces
-        .iter()
-        .filter(|i| {
-            match has_ipv6_connectivity(i, ipv6_address, network_info.ipv6_subnet, &ping_target) {
-                Ok(result) => result,
-                Err(e) => {
-                    eprintln!("Error testing connectivity on {}: {}", &i.name, e);
-                    false
-                }
+    let mut fastest_interface = None;
+
+    for interface in interfaces.iter() {
+        // Old nodes are still configured with a local IPv4 interface connection
+        // Local IPv4 interfaces must be filtered out
+        match has_ipv6_connectivity(
+            interface,
+            ipv6_address,
+            network_info.ipv6_subnet,
+            &ping_target,
+        ) {
+            Ok(true) => {
+                fastest_interface = Some(interface);
+                break;
             }
-        })
-        .collect();
+            Ok(false) => {
+                eprintln!(
+                    "Interface {} does not have IPv6 connectivity.",
+                    interface.name
+                );
+            }
+            Err(e) => {
+                eprintln!("Error testing connectivity on {}: {}", interface.name, e);
+            }
+        }
+    }
 
-    // Only assign the fastest interface to ipv6.
-    let fastest_interface = ipv6_interfaces
-        .first()
-        .context("Could not find any network interfaces")?;
+    let fastest_interface = fastest_interface
+        .context("Could not find any network interfaces with IPv6 connectivity")?;
```

Comment:
After the fastest interface with ipv6 connectivity is found, we stop checking the interfaces for an ipv6 connection. This is avoids a costly call to has_ipv6_connectivity (which can take up to 2 minutes!)

Reply:
I think we could have the same behavior using lazy iterators just by skipping the `.collect()` here, and instead calling `.next()` instead of `.first()`.

Reply:
I made this change, and then clippy told me to replace the filter-next calls with a find call, so now the diff is much cleaner!

Reply:
🧑‍🍳🤌